### PR TITLE
Mhv 61344 contact list api errors

### PR DESCRIPTION
**Pull request by KyleCardwell**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

## Summary

- added alerts for two possible errors on contact list: when saving returns an error, and when the api is not able to load the list of recipients.

### [MHV-61344](https://jira.devops.va.gov/browse/MHV-61344) - Contact list error handling-Api errors and can't save changes ###

> Contact list error handling-Api errors and can't save changes
> 
> From Joel:  
> Kyle let Emily and I know that the contact list API could fail in two ways so we need to create errors for that:
> 
> The initial call to the API fails, and a list of contacts / associations won't load.
> Calls to the API to save changes fail.
> Draft:  https://www.figma.com/design/uddjxccqvUyz5FSg1aD5wi/ZFeat---Triage-Group-Selection-Flows?node-id=1427-111430&t=bGmlt4ZkQcHyrbJJ-4
>  
> Given: a user is on the compose screen
> when: they click on the informational dropdown and navigate to the contact list
> and: the contact list API call fails
> then: an error alert is displayed
> and: the “Need help” section is hidden
> and: the “Save contact list” button is hidden and: the “Go back” button becomes the primary button
> 
> User Story: As a Secure Messaging User, I want to be understand errors that may occur when I am updating my contact list preferences
> 
> Given: User can't find a triage group/team
> 
> When: User clicks on the dropdown for help to find out what to do if user can't find a contact/team/group
> Then:  User can edit contact list from SM on VA.gov without having to navigate to Classic to do that task and an error occurs
> 
> Feature Flag Y/N ? Y-For all contact list functionality
> Manual Testing Y/N ? Y
> Automated Testing Y/N ? Y
> Accessibility Testing Y/N ? Y
> 
> UCD Validation Y/N ? Y
> 
> Acceptance Criteria:
> AC1  Error instances have been added (2)
> AC2  UCD validation is complete

## Testing done

- manual testing with forced error responses
- updated unit tests

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |Screenshot 2024-09-17 at 8 46 31 AM|
| Desktop |        |Screenshot 2024-09-17 at 8 46 05 AM|
| Mobile  |        |Screenshot 2024-09-17 at 9 33 33 AM|
| Desktop |        |Screenshot 2024-09-17 at 9 33 06 AM|

## What areas of the site does it impact?

Va.gov - Secure Messaging

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv-secure-messaging/containers/EditContactList.jsx [hunk 1]
```diff
@@ -152,6 +157,30 @@ const EditContactList = () => {
     [isMinimumSelected],
   );
 
+  const goBackButton = () => {
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/containers/EditContactList.jsx:160]: I would suggest declaring this as a component, not a function. 
```suggestion
  const GoBackButton = () => {

```

File: src/applications/mhv-secure-messaging/containers/EditContactList.jsx [hunk 2]
```diff
@@ -175,88 +203,101 @@ const EditContactList = () => {
...
+              We’re sorry. There’s a problem with our system. Try again later.
+            </p>
+            <p>
+              If it still doesn’t work, call us at{' '}
+              <VaTelephone contact={CONTACTS.MY_HEALTHEVET} /> (
+              <VaTelephone contact={CONTACTS['711']} tty />
+              ). We’re here Monday through Friday, 8:00 a.m. to 8:00 p.m. ET.
+            </p>
+          </VaAlert>
+          {goBackButton()}
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/containers/EditContactList.jsx:230]: If changed from a function to a component, render here like so 👇🏻 
```suggestion
          <GoBackButton />

```

File: src/applications/mhv-secure-messaging/containers/EditContactList.jsx [hunk 3]
```diff
@@ -175,88 +203,101 @@ const EditContactList = () => {
...
-              data-dd-action-name="Contact List Go Back Button"
-              onClick={handleCancel}
-            >
-              <div className="vads-u-margin-right--0p5">
-                <va-icon icon="navigate_far_before" aria-hidden="true" />
+                  onClick={e => handleSave(e)}
+                  data-testid="contact-list-save"
+                  data-dd-action-name="Contct List Save Button"
+                />
+                {goBackButton()}
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/containers/EditContactList.jsx:295]: Same as above
```suggestion
                <GoBackButton />

```

**Review by alex-morgun-1 — COMMENTED** [review]
Great work. Thanks for including unit tests too. Just a minor syntax consideration, left a comment below. Let me know what you think

**Review by vsaleem — APPROVED** [review]
LGTM👍

**Review by alex-morgun-1 — APPROVED** [review]
✅

**Review by powellkerry — APPROVED** [review]

**Review by KolbyVA — APPROVED** [review]

**Pull request merged**